plugin/contrib/mesos/pkg/executor/executor_test.go

```diff
@@ -385,12 +391,151 @@ func TestExecutorLaunchAndKillTask(t *testing.T) {
+
+	err := zw.Close()
+	assert.NoError(t, err)
+
+	// create fake apiserver
+	testApiServer := NewTestServer(t, api.NamespaceDefault, nil)
+	defer testApiServer.server.Close()
+
+	// temporary directory which is normally located in the executor sandbox
+	staticPodsConfigPath, err := ioutil.TempDir("/tmp", "executor-k8sm-archive")
+	assert.NoError(t, err)
+	defer os.RemoveAll(staticPodsConfigPath)
+
+	mockDriver := &MockExecutorDriver{}
+	updates := make(chan interface{}, 1024)
+	config := Config{
+		Docker:  dockertools.ConnectToDockerOrDie("fake://"),
+		Updates: make(chan interface{}, 1), // allow kube-executor source to proceed past init
+		APIClient: client.NewOrDie(&client.Config{
+			Host:    testApiServer.server.URL,
+			Version: testapi.Version(),
+		}),
+		Kubelet: &kubelet.Kubelet{},
+		PodStatusFunc: func(kl *kubelet.Kubelet, pod *api.Pod) (api.PodStatus, error) {
+			return api.PodStatus{
+				ContainerStatuses: []api.ContainerStatus{
+					{
+						Name: "foo",
+						State: api.ContainerState{
+							Running: &api.ContainerStateRunning{},
+						},
+					},
+				},
+				Phase: api.PodRunning,
+			}, nil
+		},
+		StaticPodsConfigPath: staticPodsConfigPath,
+	}
+	executor := New(config)
+	hostname := "h1"
+	go executor.InitializeStaticPodsSource(func() {
+		kconfig.NewSourceFile(staticPodsConfigPath, hostname, 1*time.Second, updates)
+	})
+
+	// create ExecutorInfo with static pod zip in data field
+	executorInfo := mesosutil.NewExecutorInfo(
+		mesosutil.NewExecutorID("ex1"),
+		mesosutil.NewCommandInfo("k8sm-executor"),
+	)
+	executorInfo.Data = buf.Bytes()
+
+	// start the executor with the static pod data
+	executor.Init(mockDriver)
+	executor.Registered(mockDriver, executorInfo, nil, nil)
+
+	// wait for static pod to start
+	seenPods := map[string]struct{}{}
+	done := make(chan struct{})
+	go func() {
+		for {
+			// filter by PodUpdate type
+			select {
+			case update, ok := <-updates:
+				if !ok {
+					return
+				}
+				switch update.(type) {
+				case kubelet.PodUpdate:
+					// register the seen pods by name
+					podUpdate := update.(kubelet.PodUpdate)
+					for _, pod := range podUpdate.Pods {
+						seenPods[pod.Name] = struct{}{}
+					}
+					if len(seenPods) == expectedStaticPodsNum {
+						close(done)
+					}
+				}
+			}
+		}
+	}()
+
+	select {
+	case <-done:
+	case <-time.After(time.Second):
+		t.Fatalf("Executor should send pod updates for %v pods, only saw %v", expectedStaticPodsNum, len(seenPods))
+	}
+
+	mockDriver.AssertExpectations(t)
+}
+
 // TestExecutorFrameworkMessage ensures that the executor is able to
 // handle messages from the framework, specifically about lost tasks
 // and Kamikaze.  When a task is lost, the executor needs to clean up
 // its state.  When a Kamikaze message is received, the executor should
 // attempt suicide.
 func TestExecutorFrameworkMessage(t *testing.T) {
+	//TODO(jdef) this test can be flaky, unexpected driver.SendStatusUpdate calls w/ TASK_LOST
```

Comment:
I've actually addressed this already in upstream_k8sm because it was breaking upstream travis-CI, TODO can be removed